### DISCUSSION TITLE
PSD layer label non-english characters encoding problem

### DISCUSSION THREAD

**Discussion by SavageStyle** [dlemstra/Magick.NET#620] [General]
```
MagickImageCollection magickImages = new MagickImageCollection(fileName)
foreach(var image in magickImages){
    var label = image.Label;
}
```
instead of Russian characters im getting question marks. "Слой-1" will be "????-1" I already read about it but found no way of fixing. Is there simple solution to this?

**Comment by dlemstra** [comment]
The labels are encoded in ANSI. The unicode version of the label is not available. I recently added a patch to fix an issue where the label was truncated. This was done by reading the unicode version of the label and only read 8 of the 16 bits available. It might be possible to store the unicode version in an attribute but I will need to investigate that. Can you share a sample file that I could use when I have to take a look at this?

**Reply by SavageStyle** [reply]
Sure, any psd file with russian layer names will do, for example
https://drive.google.com/open?id=1uSDQCfF-pQfnMEgn8FSqHiYRp_MkGjod

Layer names are actually really important, would be absolutely great if that can be fixed! Looking forward for new versions of Magick.NET

**Reply by dlemstra** [reply]
Could you write the names of the layers here? And can I use your sample file in a unit test?

**Reply by SavageStyle** [reply]
The layer names, for example, are:
Волшебник-1
Wizard-2

And image is just wizard from imagemagick.org, i guess it is yours, any image will do

**Reply by dlemstra** [reply]
I am getting the following names: `Волшебник-2` and `Wizard-1`. Not sure why the numbers are mixed up?

**Reply by dlemstra** [reply]
In the next version of Magick.NET you will be able to do this:

```C#
using ImageMagick.Formats.Psd;

var info = PsdAdditionalInfo.FromImage(image);
return info.LayerName;
```

**Reply by DeTK** [reply]
TY dlemstra!
